src/mono/mono/mini/aot-compiler.c

```diff
@@ -6589,6 +6589,8 @@ emit_and_reloc_code (MonoAotCompile *acfg, MonoMethod *method, guint8 *code, gui
 						direct_call_target = symbol;
 						patch_info->type = MONO_PATCH_INFO_NONE;
 					} else if ((m_class_get_image (patch_info->data.method->klass) == acfg->image) && !got_only && is_direct_callable (acfg, method, patch_info)) {
+						// FIXME: Currently it only works for wasm. On other platforms it fails as some callees require initialization.
+#ifdef TARGET_WASM
 						MonoCompile *callee_cfg = (MonoCompile *)g_hash_table_lookup (acfg->method_to_cfg, cmethod);
```

Comment:
Note that emit_and_reloc_code () is never used on wasm, so this shouldn't be neccessary.

Reply:
Ok, disabled for now. I've added a tracking issue for it as additional size savings might be achieved - https://github.com/dotnet/runtime/issues/82224